### PR TITLE
Use an iterator in get_sad

### PR DESCRIPTION
**Pull request by lu-zero**
- [x] write a benchmark for it
- [x] write the simplest possible iterator
- [x] benchmark
- [x] write a proper unit test for `get_sad()` (#611)
- [x] add `size_hint()` and additional functions to speed it up further

(I tested by comparing the `rav1e` output)

### CODE REVIEW

File: src/plane.rs [hunk 1]
```diff
@@ -220,12 +220,38 @@ impl Plane {
...
   pub x: isize,
   pub y: isize
 }
 
+pub struct IterWidth<'a> {
+    ps: PlaneSlice<'a>,
+    width: usize,
+}
+
+// TODO: Implement more methods
```
Comment (rzumer) [inline src/plane.rs]: Could you just clarify what methods you are referring to, here?

Comment (rzumer) [reply src/plane.rs]: OK, I was just wondering if it was worth to indicate the methods that would be useful to implement in this particular case within the comment. If we want to implement all remaining methods then I guess it is not so important.

Comment (barrbrain) [reply src/plane.rs]: If that means we no longer have a clear intent to add more methods then we should drop the TODO.

Comment (lu-zero) [reply src/plane.rs]: Done :)

**Pull request merged**